src/azul/service/manifest_service.py

```diff
@@ -361,7 +361,10 @@ def get_manifest(self,
                                                  filters=filters)
         if object_key is None:
             object_key = generator.compute_object_key()
-        presigned_url = self._get_cached_manifest(generator, object_key)
+        if partition.index < 1:
```

Comment:
Do we expect to ever observe negative numbers here or floats? If not, 
```suggestion
        if partition.index == 0:
```